pkg/controller/networkpolicy/controller_query_test.go

```diff
@@ -0,0 +1,228 @@
+	"github.com/vmware-tanzu/antrea/pkg/apis/networking/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"testing"
+	"time"
+)
+
+// pods represent kubernetes pods for testing proper query results
+var pods = []v1.Pod{
+	{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "podA",
+			Namespace: "testNamespace",
+			Labels:    map[string]string{"foo": "bar"},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{{
+				Name: "container-1",
+			}},
+			NodeName: "nodeA",
+		},
+		Status: v1.PodStatus{
+			Conditions: []v1.PodCondition{
+				{
+					Type:   v1.PodReady,
+					Status: v1.ConditionTrue,
+				},
+			},
+			PodIP: "1.2.3.4",
+		},
+	},
+	{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "podB",
+			Namespace: "testNamespace",
+			Labels:    map[string]string{"foo": "bar"},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{{
+				Name: "container-1",
+			}},
+			NodeName: "nodeA",
+		},
+		Status: v1.PodStatus{
+			Conditions: []v1.PodCondition{
+				{
+					Type:   v1.PodReady,
+					Status: v1.ConditionTrue,
+				},
+			},
+			PodIP: "1.2.3.4",
+		},
+	},
+}
+
+// polices represent kubernetes policies for testing proper query results
+//
+// policy 0: select all pods and deny default ingress
+// policy 1: select all pods and deny default egress
+
+var policies = []networkingv1.NetworkPolicy{
+	{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-ingress-egress",
+			Namespace: "testNamespace",
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{"foo": "bar"},
+			},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{
+				{
+					From: []networkingv1.NetworkPolicyPeer{
+						{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels:      map[string]string{"foo": "bar"},
+								MatchExpressions: nil,
+							},
+						},
+					},
+				},
+			},
+			Egress: []networkingv1.NetworkPolicyEgressRule{
+				{
+					To: []networkingv1.NetworkPolicyPeer{
+						{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels:      map[string]string{"foo": "bar"},
+								MatchExpressions: nil,
+							},
+						},
+					},
+				},
+			},
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeEgress,
+			},
+		},
+	},
+	{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default-deny-egress",
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{"foo": "bar"},
+			},
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeEgress,
+			},
+		},
+	},
+}
+
+var namespaces = []v1.Namespace{
+	{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "testNamespace",
+			UID:  "testNamespaceUID",
+		},
+	},
+}
+
+func makeControllerAndEndpointQueryReplier(objects ...runtime.Object) (*networkPolicyController, *EndpointQueryReplier) {
```

Comment:
do you ever use the returned `networkPolicyController` object? it does not seem so.